lib/Federation/ResourceTypeRegisterListener.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+/*
+ * @copyright Copyright (c) 2023 Joas Schilling <coding@schilljs.com>
+ *
+ * @author Joas Schilling <coding@schilljs.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Federation;
+
+use OCA\Talk\Config;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\OCM\Events\ResourceTypeRegisterEvent;
+use OCP\OCM\IOCMProvider;
+
+/**
+ * @template-implements IEventListener<Event>
+ */
+class ResourceTypeRegisterListener implements IEventListener {
+
+	public function __construct(
+		protected Config $talkConfig,
+		protected IOCMProvider $provider,
+		protected CloudFederationProviderTalk $talkProvider,
+	) {
+	}
+
+	public function handle(Event $event): void {
+		if (!$event instanceof ResourceTypeRegisterEvent) {
+			// Unrelated
+			return;
+		}
+
+		if (!$this->talkConfig->isFederationEnabled()) {
+			return;
+		}
+
+		$event->registerResourceType(
+			FederationManager::TALK_ROOM_RESOURCE,
+			$this->talkProvider->getSupportedShareTypes(),
+			[
+				'v4' => '/ocs/v2.php/apps/spreed/',
```

Comment:
@ArtificialOwl 

do you happen to know what data this is?
I assume the key is like the "what" and the value is the "where"?

so we should use something like
```suggestion
				'talk-v1' => '/ocs/v2.php/apps/spreed/api/',
```

because it's the first version of talk federation and the subsequent API is in `/ocs/v2.php/apps/spreed/api/*` ?

Reply:
From my point of view both are totally acceptable; you can set which ever you want as long as you do not need to follow a specific requirement from the author of the talk app

if you want more details about the ocm api https://cs3org.github.io/OCM-API/